pycalphad/mapping/strategy/isopleth_strategy.py

```diff
@@ -0,0 +1,286 @@
+from typing import Union
+import logging
+import itertools
+import copy
+
+import numpy as np
+
+from pycalphad import Database, variables as v
```

Comment:
Not sure exactly where this code should live, but I would like to be able to detect a pseudo binary case, where there's more than 2 components, but the tie-lines are in plane of the calculation (maybe we need to switch to binary strategy in that case?). 

Would be good if there's an example too. See https://github.com/pycalphad/pycalphad/issues/438 for some details

Reply:
Technically we don't need generalized components, because we can still have pseudo binary cases like (Hf,Ta)C or (Hf,Ta)B2 that don't require them. However, I think we should defer and merge and open a separate issue.